app/controllers/api/flavors_controller.rb

```diff
@@ -1,4 +1,24 @@
 module Api
   class FlavorsController < BaseController
+    def create_resource(_type, _id, data)
+      task_id = Flavor.create_flavor_queue(User.current_user.id, EmsCloud.find(data['ems']['id']), data)
```

Comment:
Rather than `EmsCloud.find` here, I think you may want to use `resource_search(:id, :providers, collection_class(:providers)`. Also, what do you think about supporting both `id` and `href`? Something along the lines of 
`{ "provider": { "href" : "/api/providers/:id" }` or ` { "provider": { "id": ":id" } }`. This [example](https://github.com/ManageIQ/manageiq-api/blob/master/app/controllers/api/configuration_script_sources_controller.rb#L25-L27) may help